File: api/models/HyperLink.py
from django.db import models


class HyperLink(models.Model):
    icon = models.ImageField(upload_to='hyperlink_icon/', verbose_name="图标")
    title = models.CharField(max_length=100, verbose_name="标题")
    desc = models.CharField(max_length=255, verbose_name="描述")
    url = models.URLField(verbose_name="链接")
    # 默认淡灰色, 预留长度
    color = models.CharField(max_length=16, verbose_name="颜色", default="#f2f2f2")

File: src/main/kotlin/com/coooolfan/uniboard/model/HyperLink.kt
package com.coooolfan.uniboard.model

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.GenerationType
import org.babyfish.jimmer.sql.Id

@Entity
interface HyperLink {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    val title: String

    val desc: String

    val url: String

    val color: String

    val icon: Byte
}